pandas/core/indexes/base.py

```diff
@@ -2256,15 +2256,15 @@ def intersection(self, other):
             indexer = indexer.take((indexer != -1).nonzero()[0])
         except:
             # duplicates
-            indexer = Index(other._values).get_indexer_non_unique(
-                self._values)[0].unique()
+            indexer = algos.unique1d(Index(other._values).get_indexer_non_unique(
```

Comment:
we don't use ``numpy.unique`` at all. This is ``.unique()`` on an Index, then ``np.array`` turns it into an indexer.